File: integrations/src/main/java/framework/core/entity/ThirdPartyResponseLog.java
package framework.core.entity;

import com.nivasafinance.common.annotations.NoArg;
import com.nivasafinance.common.audit.AuditableEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.javers.core.metamodel.annotation.TypeName;

import java.util.UUID;

@Entity
@Table(name = "third_party_response_log")
@TypeName("third_party_response_log")
@NoArg
@NoArgsConstructor
@AllArgsConstructor
@Data
@EqualsAndHashCode(callSuper = true)
public class ThirdPartyResponseLog extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "entity_type", length = 3, nullable = false)
    private Integer entityType;

    @Column(name = "entity_id", length = 20, nullable = true)
    private String entityId;

    @Column(name = "request_method", length = 16, nullable = false)
    private String requestMethod;

    @Column(name = "url", length = 512, nullable = false)
    private String url;

    @Column(name = "request")
    private String request;

    @Column(name = "response")
    private String response;

    @Column(name = "http_status_code", length = 4)
    private Integer httpStatusCode;

    @Column(name = "response_time_ms", length = 20)
    private Long responseTimeInMs;

    @Column(name = "provider_name", length = 50)
    private String providerName;

    @Column(name = "provider_config_id", length = 50)
    private String providerRefId;

    @Column(name = "business_purpose")
    private String businessPurpose;

    @Column(name = "business_entity")
    private String businessEntityName;

    @Column(name = "api_purpose")
    private String apiPurpose;
}

File: integrations/src/main/kotlin/framework/core/entity/ThirdPartyResponseLog.kt
package framework.core.entity

import annotations.NoArg
import audit.AuditableEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.javers.core.metamodel.annotation.TypeName
import java.util.UUID

@Entity
@Table(name = "third_party_response_log")
@TypeName("third_party_response_log")
@NoArg
@Suppress("LongParameterList")
class ThirdPartyResponseLog(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: UUID? = null,

    @Column(name = "entity_type", length = 3, nullable = false)
    var entityType: Int? = null,

    @Column(name = "entity_id", length = 20, nullable = true)
    var entityId: String? = null,

    @Column(name = "request_method", length = 16, nullable = false)
    var requestMethod: String? = null,

    @Column(name = "url", length = 512, nullable = false)
    var url: String? = null,

    @Column(name = "request")
    var request: String? = null,

    @Column(name = "response")
    var response: String? = null,

    @Column(name = "http_status_code", length = 4)
    var httpStatusCode: Int? = null,

    @Column(name = "response_time_ms", length = 20)
    var responseTimeInMs: Long? = null,

    @Column(name = "provider_name", length = 50)
    var providerName: String? = null,

    @Column(name = "provider_config_id", length = 50)
    var providerRefId: String? = null,

    @Column(name = "business_purpose")
    var businessPurpose: String? = null,

    @Column(name = "business_entity")
    var businessEntityName: String? = null,

    @Column(name = "api_purpose")
    var apiPurpose: String? = null,
) : AuditableEntity()
